reposilite-backend/src/main/kotlin/com/reposilite/console/infrastructure/ConsoleSseHandler.kt

```diff
@@ -0,0 +1,112 @@
+
+    @OpenApi(
+        // TODO: does this need better endpoint name?
+        path = "/api/console/log",
+        methods = [HttpMethod.GET],
+        // TODO: the responses parameter is *technically* not really needed, but would be nice to have
+        responses = [
+            OpenApiResponse(
+                status = "200",
+                // TODO: better description?
+                description = "Continuously sends out the log as data messages",
+                content = [OpenApiContent(from = ExecutionResponse::class)]
+            )
+        ],
+        tags = ["Console"]
+    )
+    override fun accept(sse: SseClient) {
+        sse.keepAlive()
+        sse.onClose {  ->
+            val session = users.remove(sse) ?: return@onClose
+            journalist.logger.info("CLI | ${session.identifier} closed connection")
+            journalist.unsubscribe(session.subscriberId)
+        }
+
+        when (users[sse]) {
+            null ->
+                authenticateContext(sse.ctx())
+                    .peek { identifier ->
+                        journalist.logger.info("CLI | $identifier accessed remote console")
+
+                        val subscriberId = journalist.subscribe {
+                            // TODO: do better
+                            if (!sse.terminated()) {
+                                sse.sendEvent(SSE_EVENT_NAME, it.value)
+                            } else {
+                                sse.close()
+                            }
+                        }
+
+                        users[sse] = SseSession(identifier, subscriberId)
+
+                        journalist.cachedLogger.messages.forEach { message ->
+                            sse.sendEvent(SSE_EVENT_NAME, message.value)
+                        }
+                    }
+                    .onError {
+                        journalist.logger.info("CLI | ${it.message} (${it.status})")
+                        sse.sendEvent(SSE_EVENT_NAME, it)
+                        sse.close()
+                    }
+        }
+    }
+
+    private fun authenticateContext(connection: Context): Result<String, ErrorResponse> {
+        return extractFromHeader(connection.header(Header.AUTHORIZATION))
+            .map { (name, secret) ->
+                Credentials(
+                    host = connection.getHost(),
+                    name = name,
+                    secret = secret
+                )
+            }
+            .flatMap { authenticationFacade.authenticateByCredentials(it) }
+            .filter(
+                { accessTokenFacade.hasPermission(it.identifier, AccessTokenPermission.MANAGER) },
+                { unauthorized("Unauthorized CLI access request from ${connection.getHost()}") }
+            )
+            .map { "${it.name}@${connection.getHost()}" }
+    }
+
+    private fun Context.getHost(): String =
+        header(forwardedIp.get()) ?: req().remoteAddr
```

Comment:
You can use `ctx.ip()`, it's dynamically resolved value that already does the same thing :+1: 

* Our cfg: https://github.com/dzikoysk/reposilite/blob/main/reposilite-backend/src/main/kotlin/com/reposilite/web/application/JavalinConfiguration.kt#L80